db/utility.js

```diff
@@ -25,10 +25,32 @@ const handleComment = (itemTransResult, itemCommentResult, commentResult)=>{
 
 
 const queryAllItem = async()=>{
-    let itemTransResult = await dbQuery('SELECT * FROM `Item_Trans` WHERE `lang`=\'zh\'')
-    let itemResult = await dbQuery('SELECT * FROM `Item`')
-    let commentResult = await dbQuery('SELECT * FROM `Comment`')
-    let itemCommentResult = await dbQuery('SELECT * FROM `Item_Comment`')
+    let itemTransResult
+    let itemResult
+    let itemCommentResult
+    let commentResult
+
+    try{
+        itemTransResult = await dbQuery('SELECT * FROM `Item_Trans` WHERE `lang`=\'zh\'')
+    }catch(e){
+        console.log("Fail itemTransResult", e)
+    }
+    try{
+        itemResult = await dbQuery('SELECT * FROM `Item`')
+    }catch(e){
+        console.log("Fail itemResult", e)
+    }
+    try{
+        itemCommentResult = await dbQuery('SELECT * FROM `Item_Comment`')
+    }catch(e){
+        console.log("Fail itemCommentResult", e)
+    }
+    try{
+        commentResult = await dbQuery('SELECT * FROM `Comment`')
+    }catch(e){
+        console.log("Fail commentResult", e)
+    }
+    
     let result = []
 
     let itemCommentObj = handleComment (itemTransResult, itemCommentResult, commentResult)
```

Comment:
just curious, will anything bad happen if for example **itemTransResult** is **undefined**? Because if the **try catch** block above
```
    try{
        itemTransResult = await dbQuery('SELECT * FROM `Item_Trans` WHERE `lang`=\'zh\'')
    }catch(e){
        console.log("Fail itemTransResult", e)
    }
```
throws an error you'll catch that error and print to console `console.log("Fail itemTransResult", e)`, but your variable **itemTransResult** will still be **undefined**, is it going to cause any issues?

Reply:
I think it will issue an error in handleComment, because itemTransResult will be used as an array. Maybe it will say itemTransResult has no member function.